lib/sensors/i2c_converter.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <cstdint>
+#include <optional>
+
+#include "io/i2c.hpp"
+#include <core/logger.hpp>
+
+namespace hyped::sensors {
+
+// TODOLater: Implement this properly
+/**
+ * @brief ADC (Texas Instruments) to be used for the hall-effect sensors (Farnell)
+ * @details This class is used to convert the analog signal from the hall-effect sensors to a
+ * digital signal over I2c
+ */
+class I2cConverter {
+ public:
+  static std::optional<I2cConverter> create(core::ILogger &logger);
+  ~I2cConverter();
+
+  std::uint8_t getAddress();
```

Comment:
you've got a `getAddress()` function but no `address_` class member...

Reply:
my b, this is meant to purly be skeleton code, will remove